File: Common/Checksum.cs
namespace Common;

public static class Checksum
{
    public static uint FromDirectory(DirectoryInfo directory)
    {
        if (!directory.Exists) throw new DirectoryNotFoundException($"{directory}");
        FileInfo[] files = directory.GetFiles();
        uint[] hashes = new uint[files.Length];
        for (int i = 0; i < files.Length; i++)
        {
            hashes[i] = unchecked((uint)files[i].Length/(uint)files.Length);
        }

        return hashes.Aggregate<uint, uint>(53214, (current, t) => current + t);
    }
}

File: IntegrationMod/src/main/java/com/example/examplemod/bfml/Checksum.java
package com.example.examplemod.bfml;

import java.io.File;
import java.nio.file.NotDirectoryException;

public final class Checksum {

    public static long FromDirectory(File directory) throws NotDirectoryException {
        if (!directory.exists()) throw new NotDirectoryException("{directory}");
        File[] files = directory.listFiles();
        long[] hashes = new long[files.length];
        for (int i = 0; i < files.length; i++)
        {
            hashes[i] = files[i].length()/files.length;
        }

        int result = 53214;
        for (long hash : hashes) {
            result += hash;
        }
        return result;
    }
}
